cmake/FindVecGeom.cmake

```diff
@@ -0,0 +1,51 @@
+#----------------------------------*-CMake-*----------------------------------#
+# Copyright 2022 UT-Battelle, LLC and other Celeritas developers.
+# See the top-level COPYRIGHT file for details.
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+#[=======================================================================[.rst:
+
+FindVecGeom
+-----------
+
+Find the VecGeom library and set up library linking and flags for use with
+Celeritas.
+
+#]=======================================================================]
+
+find_package(VecGeom QUIET CONFIG)
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(VecGeom CONFIG_MODE)
+
+if(VecGeom_FOUND)
+  set_target_properties(VecGeom::vecgeom PROPERTIES
+    CELERITAS_CUDA_STATIC_LIBRARY VecGeom::vecgeomcuda_static
+    CELERITAS_CUDA_MIDDLE_LIBRARY VecGeom::vecgeomcuda
+    CELERITAS_CUDA_FINAL_LIBRARY VecGeom::vecgeomcuda
+  )
+  if(CELERITAS_USE_CUDA)
```

Comment:
```suggestion
  if(CELERITAS_USE_CUDA)
    set_target_properties(VecGeom::vecgeom PROPERTIES
      CELERITAS_CUDA_STATIC_LIBRARY VecGeom::vecgeomcuda_static
      CELERITAS_CUDA_MIDDLE_LIBRARY VecGeom::vecgeomcuda
      CELERITAS_CUDA_FINAL_LIBRARY VecGeom::vecgeomcuda
    )
```

Reply:
Yup. I moved this from the main cmakelists, which had the unnecessary set. Updating.